# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pclose.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

FILE* f is freed() in fclose - thus save the pid instead of dereferencing
the freed pointed.

  - Rene Rebe <rene@exactcode.de>

--- dietlibc-0.29/libugly/pclose.c.orig	2005-06-16 13:28:05.000000000 +0200
+++ dietlibc-0.29/libugly/pclose.c	2005-06-16 13:32:08.000000000 +0200
@@ -4,8 +4,9 @@
 
 int pclose(FILE *f) {
   int status;
+  pid_t pid = f->popen_kludge;
   fclose(f);
-  if (waitpid(f->popen_kludge,&status,0)>=0)
+  if (waitpid(pid,&status,0)>=0)
     return status;
   return -1;
 }
